lib/novu/environments.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Novu.Environments do
+  @moduledoc """
+  Provide access to the Novu Environments API
+  """
+
+  alias Novu.Http
+
+  @doc """
+  Retrieve the current environment
+
+  [API Documentation](https://docs.novu.co/api-reference/environments/get-current-environment)
+  """
+  @spec get_current_environment :: Http.response()
+  def get_current_environment do
+    Http.get("/v1/environments/me")
+  end
+
+  @doc """
+  Create a new environment using it's name. Optionally set the parent of the environment to create
+
+  [API Documentation](https://docs.novu.co/api-reference/environments/create-environment)
+  """
+  @spec create_environment(name :: String.t(), opts :: Keyword.t()) :: Http.response()
+  def create_environment(name, opts \\ []) do
+    payload =
+      %{name: name}
+      |> add_optional_value(:parentId, Keyword.get(opts, :parentId))
+
+    Http.post("/v1/environments", payload)
```

Comment:
We may need to update Credo to ensure the check is correct but we try to follow this styleguide: https://github.com/christopheradams/elixir_style_guide#avoid-single-pipelines

```suggestion
    payload = add_optional_value(%{name: name}, :parentId, Keyword.get(opts, :parentId))

    Http.post("/v1/environments", payload)
```

Reply:
Uhmm, didn't know about this styleguide (first time using elixir). Do you think it would be a good idea to update Credo and fix the code where needed in this pr?

If I'm not mistaken, check is `:allow_0_arity_functions`, right?

Reply:
I would do them in a separate PR but happy to help walk you through the changes we need 😀